src/ssl/support.cc

```diff
@@ -213,14 +244,24 @@ int Ssl::matchX509CommonNames(X509 *peer_cert, void *check_data, int (*check_fun
         int numalts = sk_GENERAL_NAME_num(altnames);
```

Comment:
A few lines above, X509_NAME_get_entry() extracts (what looks like) a CN and then assumes that the extracted name should be compared as a domain name. Is it possible that the extracted name is an IP address? PR addAltNameWithSubjectCn() code looks similar and assumes the extracted NID_commonName can be an IP. If the answer is "yes, it is possible", then this PR should adjust the above code to handle IPs, right? Otherwise, please add a source code comment explaining why IP addresses are not possible in that contexts (or possible but should be ignored).

I cannot attach this change request to that currently-unmodified code.

Reply:
Yes, it's possible. The issue in my case is that even though the extracted name _is_ an IP address, it isn't a match and thus we must go through the SANs until we get a valid comparison. Your point is of course valid though, if we make `*check_func` handle both types of comparison, the code below can be simplified to just remove the skipping of non-GEN_DNS entries. I'll look into a modification of the check_func code.

Reply:
I assume that the check function is now expected to handle both IP addresses and domain names. Thus, recent PR modifications address this change request. This PR will need to _document_ that expectation (probably inside that check function type description), and there may be concerns about check function ability to correctly determine the type of information it is given, but I am resolving _this_ change request as addressed. Thank you.